Feat implement whatsapp with vonage bun

parthj000 opened the pull request:
hey @gewenyu99 @Haimantika @loks0n , i have raised the PR please review it !!

This is a video demonstrating the bun whatsapp bot with vonagee-api


[](url)

https://github.com/appwrite/templates/assets/146522413/162ff112-5585-475c-b7d6-c0a725e21f19

parthj000 commented:
![image](https://github.com/appwrite/templates/assets/146522413/c653003d-f605-4a1a-a125-682d975bc2de)

gewenyu99 commented on bun/starter/whatsapp-with-vonage/README.md:
```diff
@@ -0,0 +1,177 @@
+`VONAGE_WHATSAPP_NUMBER`
+
+`PORT`
+## Run Locally
```
Please remove these additional sections and keep this consistent with the existing Whatsapp vonage templates

gewenyu99 commented on bun/starter/whatsapp-with-vonage/src/main.js:
```diff
@@ -0,0 +1,64 @@
+import express from "express"
```
Why are you importing express? You don't need express with Appwrite functions, it's already got a node based webserver at the front

gewenyu99 reviewed (changes requested):
Hey, I see a few issues.
1. you've nested this under bun/starter, which is not entirely correct
2. You've created a standalone express server, this should be an [Appwrite Function](https://appwrite.io/docs/products/functions)

Thanks!

parthj000 commented:
@loks0n i have done all the required changes !!

gewenyu99 commented:
Hey,

Due to time constraints, I'm going to mark this PR `hacktoberfest-accepted` for now so you get DO's Hacktoberfest rewards. We'll continue to work with you on this issue for review and merge.

When it is merged, we'll contact you for Appwrite-specific Hacktoberfest swag.

Thanks for helping us improve Appwrite!

loks0n commented on bun/starter/src/main.ts:
```diff
@@ -25,9 +25,9 @@ export default async ({ req, res, log, error }: any) => {
   // `res.json()` is a handy helper for sending JSON
   return res.json({
-    motto: "Build Fast. Scale Big. All in One Place.",
+    motto: "Build like a team of hundreds_",
```
Please revert this change.

loks0n commented on bun/starter/package.json:
Please revert this change.

loks0n replied:
Reset this file please, there is a whitespace changegit checkout origin/main bun/starter/README.md

loks0n commented on bun/starter/bun.lockb:
Please revert this change.

loks0n commented on bun/starter/README.md:
Please revert this change.

loks0n commented on bun/starter/.gitignore:
Please revert this change.

loks0n commented on bun/whatsapp-with-vonage/README.md:
```diff
@@ -0,0 +1,126 @@
+| ------------- | ---------------------------------- | -------- | ------------------- | -------------------- |
+| Status  | showing the status of message    | body  | string| submitted/read/delievered          |
+
+
```
Can you remove all the double empty lines in this file?

loks0n commented on bun/whatsapp-with-vonage/README.md:
```diff
@@ -0,0 +1,126 @@
+
+| Name          | Description                        | Location | Type                | Sample Value         |
+| ------------- | ---------------------------------- | -------- | ------------------- | -------------------- |
+| Status  | showing the status of message    | body  | string| submitted/read/delievered          |
```
Please format the file with prettier

loks0n commented on bun/whatsapp-with-vonage/README.md:
```diff
@@ -0,0 +1,126 @@
+| ----------------- | ------------------------ |
+| Runtime           | Bun (1.0)              |
+| Entrypoint        | `src/main.js`            |
+| Build Commands    | `Bun install src/main.js` |
```
This is not correct, please see the bun starter as an example

loks0n commented on bun/whatsapp-with-vonage/README.md:
```diff
@@ -0,0 +1,126 @@
+| Setting           | Value                    |
+| ----------------- | ------------------------ |
+| Runtime           | Bun (1.0)              |
+| Entrypoint        | `src/main.js`            |
```
This is not correct, please see the bun starter as an example

loks0n commented on bun/whatsapp-with-vonage/README.md:
```diff
@@ -0,0 +1,126 @@
+| Timeout (Seconds) | 15                       |
+
+
+## ENVIRONMENT VARIABLES
+
+Following environment variables are required :
+
+`VONAGE_API_KEY`
+
+`VONAGE_ACCOUNT_SECRET`
+
+`VONAGE_SIGNATURE_SECRET`
+
+`VONAGE_WHATSAPP_NUMBER`
+
+## Documentation
+
+[Vonage Documentation ](https://developer.vonage.com/en/documentation)
+
```
We already mention this above, please remove this

loks0n commented on bun/whatsapp-with-vonage/public/index.html:
Can you use the HTML from the `node/whatsapp-with-vonage` function and put it in `static` folder please

loks0n commented on bun/whatsapp-with-vonage/src/main.js:
```diff
@@ -0,0 +1,70 @@
+import jwt from 'jsonwebtoken';
+
```
There are loads of random empty lines in the this file... please remove them.

loks0n commented on bun/whatsapp-with-vonage/src/main.js:
```diff
@@ -0,0 +1,70 @@
+
+export default async ({ req, res, log, error }) => {
+
+    const html = file("public/index.html")
```
This won't work, please check out how the other bun templates are doing it.

loks0n commented on bun/whatsapp-with-vonage/src/main.js:
```diff
@@ -0,0 +1,70 @@
+import jwt from 'jsonwebtoken';
+
+
+export default async ({ req, res, log, error }) => {
```
Try structuring the code in a similar way to the node template.

loks0n commented on bun/whatsapp-with-vonage/src/main.js:
```diff
@@ -0,0 +1,70 @@
+
+        };
+        if (!(req.body.text == null)) {
+            await fetch("https://messages-sandbox.nexmo.com/v1/messages", requestOptions)
```
You need to import fetch from `undici`

loks0n commented on bun/whatsapp-with-vonage/src/main.js:
```diff
@@ -0,0 +1,70 @@
+        })
+
+        const myHeaders = new Headers();
+        myHeaders.append("Accept", "application/json");
```
This header is not required

loks0n commented on bun/whatsapp-with-vonage/src/main.js:
```diff
@@ -0,0 +1,70 @@
+        myHeaders.append("Content-Type", "application/json");
+        myHeaders.append("Authorization", `Basic ${basicAuthToken}`);
+
+        const raw = JSON.stringify({
```
raw is not a very descriptive name

parthj000 commented:
@loks0n , i have done the required changes !

loks0n commented on bun/starter/README.md:
````diff
@@ -26,7 +26,7 @@ Sample `200` Response:
 ```json
 {
-  "motto": "Build Fast. Scale Big. All in One Place.",
+  "motto": "Build like a team of hundreds_",
````
Please revert this change

loks0n commented on bun/starter/.gitignore:
Revert this change

loks0n commented on bun/starter/README.md:
```diff
@@ -45,4 +45,4 @@ Sample `200` Response:
 ## 🔒 Environment Variables
 
-No environment variables required.
+No environment variables required.
```
Please revert this change

loks0n replied:
Reset this file please, there is a whitespace change`git checkout origin/main bun/starter/README.md`

loks0n commented on bun/starter/package.json:
```diff
@@ -13,4 +13,4 @@
     "prettier": "^3.0.0"
   }
-}
+}
```
Please revert this change

loks0n commented on bun/starter/bun.lockb:
Please revert this change

loks0n commented on bun/starter/src/main.ts:
```diff
@@ -1,4 +1,4 @@
-import { Client } from 'node-appwrite';
+import { Client } from "node-appwrite";
```
Please revert this change

loks0n commented on bun/whatsapp-with-vonage/README.md:
```diff
@@ -0,0 +1,102 @@
+| ----------------- | -------------- |
+| Runtime           | Bun (1.0)      |
+| Entrypoint        | `src/main.ts`  |
+| Build Commands    | `Bun install ` |
```
Should be lowercase `bun install` here

loks0n commented on bun/whatsapp-with-vonage/src/main.ts:
```diff
@@ -0,0 +1,74 @@
+    });
+  }
+
+  if (req.method === "POST") {
```
This if state is not required

loks0n commented on bun/whatsapp-with-vonage/src/main.ts:
```diff
@@ -0,0 +1,74 @@
+      body: bodyContent,
+    };
+    if (!(req.body.text == null)) {
+      await fetch(
```
Can we just do all this inline:
```js
  await fetch(`https://messages-sandbox.nexmo.com/v1/messages`, {
    method: 'POST',
    body: JSON.stringify({
      from: process.env.VONAGE_WHATSAPP_NUMBER,
      to: req.body.from,
      message_type: 'text',
      text: `Hi there! You sent me: ${req.body.text}`,
      channel: 'whatsapp',
    }),
    headers: {
      'Content-Type': 'application/json',
      Authorization: `Basic ${basicAuthToken}`,
    },
  });
```

loks0n commented on bun/whatsapp-with-vonage/src/main.ts:
```diff
@@ -0,0 +1,74 @@
+  }
+
+  if (req.method === "POST") {
+    const secret = `${Bun.env.VONAGE_API_KEY}:${Bun.env.VONAGE_ACCOUNT_SECRET}`;
+    const basicAuthToken = btoa(secret);
+    const authHeader = req.headers.authorization.split(" ")[1];
+    await jwt.verify(
+      authHeader,
+      Bun.env.SIGNATURE_SECRET,
+      { algorithms: ["HS256"] },
+      (error, decodedToken) => {
+        if (error) {
+          return res.json(
+            {
+              ok: false,
+              error: "can't verify",
+            },
+            400
+          );
+        }
+      }
+    );
```
This is quite complicated. Take a look the [node template](https://github.com/appwrite/templates/blob/6dab35f1f221ba075640d93f5a8e374f4fba4349/node/whatsapp-with-vonage/src/main.js#L20)
```js
  const token = (req.headers.authorization ?? '').split(' ')[1];
  var decoded = jwt.verify(token, process.env.VONAGE_API_SIGNATURE_SECRET, {
    algorithms: ['HS256'],
  });

  if (sha256(req.bodyRaw) != decoded['payload_hash']) {
    return res.json({ ok: false, error: 'Payload hash mismatch.' }, 401);
  }
```

parthj000 commented:
@loks0n 
I m little confused which changes you want me to revert .
Besides it i have added payload hash verification , and also added the asked changes, can you review them !

loks0n commented on bun/starter/bun.lockb:
Again reset this file

loks0n commented on bun/whatsapp-with-vonage/src/main.ts:
```diff
@@ -0,0 +1,67 @@
+import { fetch } from "undici";
+
+export default async ({ req, res, log, error }) => {
+  if (req.method === "GET") {
+    const fileName = Bun.resolveSync("../static/index.html", import.meta.dir);
+    const html = await Bun.file(fileName).text();
+
+    return res.send(html, 200, {
+      "Content-Type": "text/html; charset=utf-8",
+    });
+  }
```
We use a utils file for this is other templates, please look

loks0n commented on bun/whatsapp-with-vonage/src/main.ts:
```diff
@@ -0,0 +1,67 @@
+    });
+  }
+
+  try {
+    const hasher = new Bun.CryptoHasher("sha256");
+    const hashedBuffer = hasher.update(JSON.stringify(req.body)).digest("hex");
+    const hashedValue = hashedBuffer.toString();
+
+    const authHeader = req.headers.authorization.split(" ")[1];
+    const decodedToken = jwt.verify(
+      authHeader,
+      Bun.env.VONAGE_SIGNATURE_SECRET,
+      {
+        algorithms: ["HS256"],
+      }
+    );
+    if (hashedValue != decodedToken["payload_hash"]) {
+      return res.json({
+        ok: false,
+        error: "Payload mismatched",
+      });
+    }
+  } catch (err) {
+    return res.json({
+      ok: false,
+      error: "can't verify",
+    });
+  }
```
Please use the same flow as the existing node template. This is completely different

loks0n commented on bun/whatsapp-with-vonage/src/main.ts:
```diff
@@ -0,0 +1,67 @@
+  const secret = `${Bun.env.VONAGE_API_KEY}:${Bun.env.VONAGE_ACCOUNT_SECRET}`;
+  const basicAuthToken = btoa(secret);
+
+  if (!(req.body.text == null)) {
+    await fetch(`https://messages-sandbox.nexmo.com/v1/messages`, {
+      method: "POST",
+      body: JSON.stringify({
+        from: Bun.env.VONAGE_WHATSAPP_NUMBER,
+        to: req.body.from,
+        message_type: "text",
+        text: `You sent me: ${req.body.text}`,
+        channel: "whatsapp",
+      }),
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: `Basic ${basicAuthToken}`,
+      },
+    });
+  }
```
Swap the case round and use early return here please https://refactoring.guru/replace-nested-conditional-with-guard-clauses

loks0n commented:
> @loks0n I m little confused which changes you want me to revert . Besides it i have added payload hash verification , and also added the asked changes, can you review them !

Please follow the Node template as closely as possible.

loks0n commented on bun/whatsapp-with-vonage/src/main.ts:
```diff
@@ -0,0 +1,65 @@
+    },
+  });
+
+  return res.json(
+    {
+      ok: true,
+    },
+    200
+  );
```
Formatting looks weird, please run prettier

loks0n commented on bun/whatsapp-with-vonage/src/main.ts:
```diff
@@ -0,0 +1,65 @@
+    );
+  }
+
+  const secret = `${Bun.env.VONAGE_API_KEY}:${Bun.env.VONAGE_ACCOUNT_SECRET}`;
```
This variable is unnecessary please match https://github.com/appwrite/templates/blob/6dab35f1f221ba075640d93f5a8e374f4fba4349/node/whatsapp-with-vonage/src/main.js#L35

loks0n commented on bun/whatsapp-with-vonage/src/main.ts:
```diff
@@ -0,0 +1,65 @@
+
+export default async ({ req, res, log, error }) => {
+  if (req.method === "GET") {
+    return res.send(getStaticFiles, 200, {
+      "Content-Type": "text/html; charset=utf-8",
+    });
```
Why is it called `getStaticFiles` when it returns the contents of a single file?

Please make a function of format like `getStaticFile(filename: string)` as like https://github.com/appwrite/templates/blob/6dab35f1f221ba075640d93f5a8e374f4fba4349/node/whatsapp-with-vonage/src/main.js#L15

loks0n commented on bun/whatsapp-with-vonage/src/main.ts:
```diff
@@ -0,0 +1,65 @@
+import { fetch } from "undici";
+import { getStaticFiles } from "./utils";
+
+export default async ({ req, res, log, error }) => {
```
We need to throw if the environment variables are not set like here https://github.com/appwrite/templates/blob/6dab35f1f221ba075640d93f5a8e374f4fba4349/node/whatsapp-with-vonage/src/main.js#L7

loks0n commented on bun/whatsapp-with-vonage/src/main.ts:
```diff
@@ -0,0 +1,65 @@
+    });
+  }
+
+  const hasher = new Bun.CryptoHasher("sha256");
+  const hashedBuffer = hasher.update(JSON.stringify(req.body)).digest("hex");
+  const hashedValue = hashedBuffer.toString();
+
+  const authHeader = req.headers.authorization.split(" ")[1];
+  const decodedToken = jwt.verify(authHeader, Bun.env.VONAGE_SIGNATURE_SECRET, {
+    algorithms: ["HS256"],
+  });
+
+  if (hashedValue != decodedToken["payload_hash"]) {
+    return res.json(
+      {
+        ok: false,
+        error: "Payload hash mismatched",
+      },
+      401
+    );
+  }
```
This is very different to the node template, please match the node template as closely as possible but using Bun utilites.
https://github.com/appwrite/templates/blob/6dab35f1f221ba075640d93f5a8e374f4fba4349/node/whatsapp-with-vonage/src/main.js#L20

loks0n commented on bun/whatsapp-with-vonage/src/utils.ts:
```diff
@@ -0,0 +1,5 @@
+export async function getStaticFiles() {
```
Please copy other bun templates to get the best version of this function

loks0n reviewed (changes requested):
Hi, 

I'm spending a lot of time going back and forth with this PR - please try to address all my comments before resubmitting for review.

loks0n commented on bun/whatsapp-with-vonage/src/utils.ts:
```diff
@@ -0,0 +1,21 @@
+//  Returns the contents of a file in the static folder
+
```
Please change this comment to match the other TS templates https://github.com/appwrite/templates/blob/6dab35f1f221ba075640d93f5a8e374f4fba4349/deno/whatsapp-with-vonage/src/utils.ts#L2

loks0n commented on bun/whatsapp-with-vonage/src/utils.ts:
```diff
@@ -0,0 +1,21 @@
+  return html;
+}
+
+// Throws an error if any of the keys are missing from the object
```
https://github.com/appwrite/templates/blob/6dab35f1f221ba075640d93f5a8e374f4fba4349/deno/whatsapp-with-vonage/src/utils.ts#L2

loks0n commented on bun/whatsapp-with-vonage/src/main.ts:
```diff
@@ -0,0 +1,66 @@
+import { fetch } from "undici";
+import { getStaticFiles, throwIfMissing } from "./utils.ts";
+
+export default async ({ req, res, log, error }: any) => {
```
Can we add basic types here as shown here: https://github.com/appwrite/templates/blob/6dab35f1f221ba075640d93f5a8e374f4fba4349/deno/whatsapp-with-vonage/src/main.ts#L6C1-L12C1

loks0n commented on bun/whatsapp-with-vonage/README.md:
````diff
@@ -0,0 +1,111 @@
+
+```json
+{
+  "ok": 200,
+  "status": "delievered/submitted/read"
````
This is not what you actually return.

loks0n commented on bun/whatsapp-with-vonage/src/main.ts:
```diff
@@ -0,0 +1,66 @@
+    `${Bun.env.VONAGE_API_KEY}:${Bun.env.VONAGE_ACCOUNT_SECRET}`
+  );
+
+  if (req.body.text == null) {
+    return res.json({ ok: true, status: req.body.status }, 200);
+  }
```
This is not necessary, please remove this

loks0n commented on bun/whatsapp-with-vonage/src/main.ts:
```diff
@@ -0,0 +1,66 @@
+      "Content-Type": "text/html; charset=utf-8",
+    });
+  }
+  const authHeader: string = (req.headers.authorization ?? "").split(" ")[1];
+  const decodedToken = jwt.verify(authHeader, Bun.env.VONAGE_SIGNATURE_SECRET, {
+    algorithms: ["HS256"],
+  });
```
Please add a space before this section and use structure similar to here
https://github.com/appwrite/templates/blob/6dab35f1f221ba075640d93f5a8e374f4fba4349/node/whatsapp-with-vonage/src/main.js#L20

First assign `token` variable

Then assign `decoded` variable

loks0n commented on bun/whatsapp-with-vonage/src/main.ts:
```diff
@@ -0,0 +1,66 @@
+  try {
+    throwIfMissing(decodedToken, ["payload_hash"]);
+  } catch (err) {
+    return res.json({ ok: false, error: err.message }, 400);
```
This error is undocumented in the README

parthj000 commented:
@loks0n , but what will happen to the event webhook ,If I remove this then incoming event webhooks will not be addressed and in documentation it is written that the webhook must be responded with 2XX response and if not done the request will be retried ... so will it not cause any problem and is it efficient to not account the webhooks giving us the status of message ?
And I have removed "text" as a parameter from the throwIfMissing function .

Documentation -https://developer.vonage.com/en/api/messages-olympus#websockets

parthj000 commented:
Please review this ! @loks0n

loks0n commented on bun/whatsapp-with-vonage/README.md:
````diff
@@ -0,0 +1,118 @@
+}
+```
+
+```json
+{
+  "ok": false,
+  "error": "Missing required parameter: payload_hash"
+}
+```
````
Let's just remove this example to match the other readmes

loks0n commented on bun/whatsapp-with-vonage/README.md:
```diff
@@ -0,0 +1,118 @@
+
+HTML form interacting with the function.
+
+## POST/
```
Please add a space between POST and /

loks0n commented on bun/whatsapp-with-vonage/README.md:
```diff
@@ -0,0 +1,118 @@
+
+## 🧰 Usage
+
+## GET/
```
Please add a space between POST and /

loks0n commented on bun/whatsapp-with-vonage/README.md:
````diff
@@ -0,0 +1,118 @@
+}
+```
+
+## POST/
+
+Event webhooks sent by vonage to update about the status of message
+
+| Name   | Description                   | Location | Type   | Sample Value              |
+| ------ | ----------------------------- | -------- | ------ | ------------------------- |
+| Status | showing the status of message | body     | string | submitted/read/delievered |
+
+**Response:**
+
+```json
+{
+  "ok": 200,
+  "status": "delievered"
+}
+```
````
Let's remove this section. We have already covered it.

loks0n commented on bun/whatsapp-with-vonage/src/main.ts:
```diff
@@ -0,0 +1,81 @@
+};
+
+export default async ({ req, res, log, error }: Context) => {
+  
```
Please remove the newline here

loks0n commented on bun/whatsapp-with-vonage/src/main.ts:
```diff
@@ -0,0 +1,81 @@
+    "VONAGE_WHATSAPP_NUMBER",
+    "VONAGE_SIGNATURE_SECRET",
+  ]);
+  
```
Please remove the double newline here and replace with a single newline

loks0n commented on bun/whatsapp-with-vonage/src/main.ts:
```diff
@@ -0,0 +1,81 @@
+  const decodedToken = jwt.verify(token, Bun.env.VONAGE_SIGNATURE_SECRET, {
+    algorithms: ["HS256"],
+  });
+  
```
Please remove the double newline here, and replace with a single new line

loks0n commented on bun/whatsapp-with-vonage/src/main.ts:
```diff
@@ -0,0 +1,81 @@
+  
+  const hasher = new Bun.CryptoHasher("sha256");
+  const hashedValue = hasher.update(req.bodyRaw).digest("hex").toString();
+  
```
Please remove the double newline here and replace with a single newline

loks0n commented on bun/whatsapp-with-vonage/src/main.ts:
```diff
@@ -0,0 +1,81 @@
+  if (hashedValue != decodedToken["payload_hash"]) {
+    return res.json({ ok: false, error: "Payload hash mismatched" }, 401);
+  }
+  
```
Please remove the double newline here and replace with a single newline

loks0n commented on bun/whatsapp-with-vonage/src/main.ts:
```diff
@@ -0,0 +1,81 @@
+  } catch (err) {
+    return res.json({ ok: false, error: err.message }, 400);
+  }
+  
```
Please remove the double newline here and replace with a single newline

loks0n commented on bun/whatsapp-with-vonage/src/main.ts:
```diff
@@ -0,0 +1,81 @@
+  if (req.body.text == null) {
+    return res.json({ ok: true, status: req.body.status }, 200);
+  }
+  
```
Please remove the double newline here and replace with a single newline

loks0n commented on bun/whatsapp-with-vonage/src/utils.ts:
```diff
@@ -0,0 +1,25 @@
+/*
+ * Returns the contents of a file in the static folder
+ */
+
```
Please remove the newline here

loks0n commented on bun/whatsapp-with-vonage/static/index.html:
```diff
@@ -0,0 +1,34 @@
+<!doctype html >
```
Please remove the space here <!doctype html >
```suggestion
<!doctype html>
```

loks0n commented on bun/whatsapp-with-vonage/src/utils.ts:
```diff
@@ -0,0 +1,25 @@
+
+export async function getStaticFiles(fileName: string) {
+  const filePath = Bun.resolveSync(`../static/${fileName}`, import.meta.dir);
+  const html = await Bun.file(filePath).text();
+  return html;
```
Let's remove the inline variable and just `return await Bun.file(filePath).text()`

loks0n commented on bun/whatsapp-with-vonage/src/utils.ts:
```diff
@@ -0,0 +1,25 @@
+ * Returns the contents of a file in the static folder
+ */
+
+export async function getStaticFiles(fileName: string) {
```
Can we rename this `getStaticFile`

parthj000 commented:
updated !

loks0n commented on bun/whatsapp-with-vonage/src/main.ts:
```diff
@@ -0,0 +1,74 @@
+    return res.json({ ok: false, error: err.message }, 400);
+  }
+
+  if (req.body.text == null) {
+    return res.json({ ok: true, status: req.body.status }, 200);
+  }
```
Please remove this.

gewenyu99 commented:
Hey there! There were a lot of big PRs during this Hacktoberfest, and we wanted to give everyone ample time to collaborate with our engineering team. If you were able to merge your PRs during October, amazing. If it’s still not merged, don’t worry about it either. Either way, we’ve got your Hacktoberfest swag minted and ready to ship. 

Please comment with your Discord username here so we can contact you about your shipping information to deliver your Hacktoberfest swag.

parthj000 commented:
@gewenyu99 ,
pars0942

> Hey there! There were a lot of big PRs during this Hacktoberfest, and we wanted to give everyone ample time to collaborate with our engineering team. If you were able to merge your PRs during October, amazing. If it’s still not merged, don’t worry about it either. Either way, we’ve got your Hacktoberfest swag minted and ready to ship.
> 
> Please comment with your Discord username here so we can contact you about your shipping information to deliver your Hacktoberfest swag.

parthj000 commented:
Hey , @gewenyu99,  I have not been contacted for the swags yet :)

adityaoberai commented:
> @gewenyu99 , pars0942
> 
> > Hey there! There were a lot of big PRs during this Hacktoberfest, and we wanted to give everyone ample time to collaborate with our engineering team. If you were able to merge your PRs during October, amazing. If it’s still not merged, don’t worry about it either. Either way, we’ve got your Hacktoberfest swag minted and ready to ship.
> > Please comment with your Discord username here so we can contact you about your shipping information to deliver your Hacktoberfest swag.

Hey Parth
Just sent you a friend request on DIscord. If you can accept it and DM me, I'll send over the form.
